cmake/HPX_AddModule.cmake

```diff
@@ -152,34 +153,30 @@ function(add_hpx_module name)
   endif()
   hpx_export_targets(hpx_${name})
 
+  # Install the headers from the source
   install(
     DIRECTORY include/hpx
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     COMPONENT ${name})
 
+  # Install the compatibility headers from the source
   if(HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS)
     install(
       DIRECTORY include_compatibility/hpx
       DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
       COMPONENT ${name})
   endif()
 
-  # Installing the global header file
-  if (${${name}_GLOBAL_HEADER_GEN})
-    install(
-      FILES ${global_header}
-      DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hpx
-      COMPONENT ${name}
-    )
-  endif()
+  # Installing the generated header files from the build dir
+  install(
+    DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/include/hpx
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/hpx
+    COMPONENT ${name}
+  )
   
   write_config_defines_file(
     NAMESPACE ${name_upper}
-    FILENAME "${CMAKE_BINARY_DIR}/hpx/${name}/config/defines.hpp")
-
-  write_config_defines_file(
-    NAMESPACE ${name_upper}
-    FILENAME "${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/hpx/${name}/config/defines.hpp")
+    FILENAME "${CMAKE_CURRENT_BINARY_DIR}/include/hpx/${name}/config/defines.hpp")
```

Comment:
Does this has to come before the `install()` above in order to be properly picked up? I'm not sure...

Reply:
Yep sorry!